base/rational.jl

```diff
@@ -10,23 +10,34 @@ struct Rational{T<:Integer} <: Real
     num::T
     den::T
 
-    function Rational{T}(num::Integer, den::Integer) where T<:Integer
-        num == den == zero(T) && __throw_rational_argerror_zero(T)
-        num2, den2 = divgcd(num, den)
-        if T<:Signed && signbit(den2)
-            den2 = -den2
-            signbit(den2) && __throw_rational_argerror_typemin(T)
-            num2 = -num2
-        end
-        return new(num2, den2)
+    # Unexported inner constructor of Rational that bypasses all checks
+    global unsafe_rational(::Type{T}, num, den) where {T} = new{T}(num, den)
```

Comment:
Since this is a private function shouldn't  be renamed? 
```
_unsafe_rational
```

Reply:
@JeffreySarnoff seemed to have some reluctance against this kind of naming convention back in https://github.com/JuliaLang/julia/pull/35492#issuecomment-618835539, although I actually thought it was indeed standard for internal functions. I don't have a strong opinion either way.

Reply:
We use the underscore convention in other parts of the code base to denote private internal function.

I'll let @StefanKarpinski  decide best.

So you don't have to keep updating PRs back and forth :)

Reply:
I think this is fine. There are lots of unexported functions that don't start with `_`.